### ISSUE TITLE
Questions about loss_cls is Nan

### ISSUE THREAD

**Issue by Simon-Stma** [GuangxingHan/Meta-Faster-R-CNN#7]
When debugging the code, I suspect that some parameters may not be understood, leading to the result that loss_cls is Nan in the first iteration. stage_1.yaml", what is the role of __BASE__ class, I see its contents are in "Base-FSOD-C4.yaml", which mentions some settings of FsodRCNN parameters, but I would like to ask in the first stage is for meta-RPN and meta-classifier training, should this FsodRCNN be the second stage? If so, why is there a new configuration file "faster_rcnn_with_fpn_pascalvoc_split1_base_classes_branch.yaml" for the second stage? Because I found that in "base-fsod-c4.yaml", DATASETS is coco dataset (the file "base-fsod-c4.yaml" only appeared in the relevant VOC dataset) I'm a little confused by the setting of these parameters, and the setting of num_classes to 1 in ROI_head.

**Comment by GuangxingHan** [comment]
This is a great point. Our explanations are as follow.

- In stage 1, we use the base classes to perform meta-training, and the idea is to sample multiple FSOD episodes using the base-classes dataset for training the Meta-RPN, Meta-Classifier and the feature backbone. The feature backbone is initialized from the imagenet pre-trained weight. The Meta-RPN and Meta-Classifier are the newly-introduced modules.

  Our Meta-Classifier takes the proposal features and only one support feature as input at one time. Thus the num_classes is set to be 1 in ROI_head, which also means that we use a binary classifier as the output.

- The base class branch training is after the meta-training. We find that reusing the meta-trained feature backbone and only learning a small detection head for base classes, can achieve comparable or slightly higher results, compared with the strong TFA baseline.

- If we first train the traditional Faster R-CNN detection model using base class, and then do the meta-training based on the learned feature backbone, we find that it have poor performance for meta-testing, using the similar 3-stage meta-training script as in this repo. We argue that the training order matters, and the reason is that the traditional softmax based detection training is more likely to be overfit to base classes and is not suitable for meta-testing which needs strong generalization ability to unseen novel classes. Moreover, simply training the base class detection head after meta-training can easily achieve high performance for base classes, which also demonstrates the idea.
